clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3978,14 +3978,24 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
                          const ArgList &Args, const char *LinkingOutput) const {
   const auto &TC = getToolChain();
   const llvm::Triple &RawTriple = TC.getTriple();
-  const llvm::Triple &Triple = TC.getEffectiveTriple();
-  const std::string &TripleStr = Triple.getTriple();
+  llvm::Triple Triple = TC.getEffectiveTriple();
 
   bool KernelOrKext =
       Args.hasArg(options::OPT_mkernel, options::OPT_fapple_kext);
   const Driver &D = TC.getDriver();
   ArgStringList CmdArgs;
 
+  // -fsycl-device-only implies a SPIRV arch triple.  Do not set if current
+  // effective triple is SYCLDevice
+  if (Args.hasArg(options::OPT_fsycl_device_only) &&
+      Triple.getEnvironment() != llvm::Triple::SYCLDevice) {
+    const char *SYCLTargetArch = "spir64";
+    if (C.getDefaultToolChain().getTriple().getArch() == llvm::Triple::x86)
+      SYCLTargetArch = "spir";
```

Comment:
How come a `const` pointer re-assignable? 0_o
Just asking, though. In case there's something wrong about this, a ternary and/or a `StringRef` present other options.

Reply:
I'm changing the pointer with the assignment.  `SYCLTargetArch` wouldn't be modifiable with something like `char * const`